sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -763,35 +764,31 @@ public Mono<Response<ShareDirectoryInfo>> setPropertiesWithResponse(FileSmbPrope
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<ShareDirectoryInfo>> setPropertiesWithResponse(ShareDirectorySetPropertiesOptions options) {
         try {
-            return withContext(context -> setPropertiesWithResponse(options.getSmbProperties(),
-                options.getFilePermissions().getPermission(), options.getFilePermissions().getPermissionFormat(),
-                context));
+            ShareFilePermission filePermission
+                = options.getFilePermissions() == null ? new ShareFilePermission() : options.getFilePermissions();
+            return withContext(
+                context -> setPropertiesWithResponse(options.getSmbProperties(), filePermission.getPermission(),
+                    filePermission.getPermissionFormat(), options.getPosixProperties(), context));
         } catch (RuntimeException ex) {
             return monoError(LOGGER, ex);
         }
     }
 
     Mono<Response<ShareDirectoryInfo>> setPropertiesWithResponse(FileSmbProperties smbProperties, String filePermission,
-        FilePermissionFormat filePermissionFormat, Context context) {
-
-        FileSmbProperties properties = smbProperties == null ? new FileSmbProperties() : smbProperties;
+        FilePermissionFormat filePermissionFormat, FilePosixProperties posixProperties, Context context) {
+        context = context == null ? Context.NONE : context;
```

Comment:
Can we add parenthesis here for better readability of code?

for ex : context = (context == null) ? Context.NONE : context;

Reply:
[link to the reply above](https://github.com/Azure/azure-sdk-for-java/pull/43112#discussion_r1890645951)